app/lib/email.server.ts

```diff
@@ -89,6 +91,42 @@ function maybeThrowSES(e: any, warning: string) {
   ])
 }
 
+/**
+ * Amazon SES has a byte limit of 262144 for TemplateData arguments.
+ * If the circular body is over this size limit, the message will be truncated.
+ * If the body is truncated, the circular website link will be removed.
+ * A warning that the email has been truncated is added and the circular website link is added back in.
+ * If truncated, the email body is aggressively truncated so there are over 60,000 bytes left for additonal
+ * escape characters added when converted to json.
+ */
+export function truncateEmailBodyIfNecessary({
+  body,
+  subject,
+  circularId,
+}: {
+  body: string
+  subject: string
+  circularId?: number
+}) {
+  const textBody = getBody(body)
+  const jsonTemplateData = JSON.stringify({
+    subject,
+    body: textBody,
+  })
+
+  if (Buffer.byteLength(jsonTemplateData, 'utf-8') <= 262144) return textBody
```

Comment:
The maximum length in the check here must match the maximum length in the call to `truncate`, otherwise the corner cases may still occur.

```suggestion
  if (Buffer.byteLength(jsonTemplateData, 'utf-8') <= 200000) return textBody
```

Even better would be to define a constant, so that you don't have to repeat the literal value.